Django role api changes

SebastiaanZ opened the pull request:
This adds support for the new position field of the Role model proposed in python-discord/site#236 targeting the `django` branch of site, by making sure roles can be deleted and the role's hierarchy position is synchronized with the database.

This PR includes support for deleting roles that were removed from the Guild.

MarkKoz commented on bot/cogs/sync/syncers.py:
```diff
@@ -224,4 +232,4 @@ def get_users_for_sync(
         )
 
-    return (len(users_to_create), len(users_to_update))
+    return len(users_to_create), len(users_to_update), None
```
Making it an integer like the others in the tuple will just make life easier.

```suggestion
    return len(users_to_create), len(users_to_update), 0
```

SebastiaanZ replied:
I figured that would make it more confusing instead. Not here, but in the logs. See the other comment.

SebastiaanZ commented on bot/cogs/sync/cog.py:
```diff
@@ -30,54 +30,68 @@ def __init__(self, bot):
-                    syncer_name, total_created, total_updated
-                )
+                total_created, total_updated, total_deleted = await syncer(self.bot, guild)
+                if total_deleted is None:
```
I've considered that, but if we do that, the logs will suggest that the user syncer is capable of deleting entries from the database, while it is not.

I figured clarity in the logs, which are read far more often, was worth more than the cost in added complexity of a single if-statement.

I don't think the if-condition makes the code complex or unclear.

MarkKoz replied:
> I've considered that, but if we do that, the logs will suggest that the user syncer is capable of deleting entries from the database, while it is not.

I hadn't considered that.

>I don't think the if-condition makes the code complex or unclear.
I don't really think it does either. It was more of a "if we can do this change for free, then why not". But turns out we can't.

SebastiaanZ commented on bot/cogs/sync/cog.py:
```diff
@@ -30,54 +30,68 @@ def __init__(self, bot):
-            role.name, role.id
-        )
+        """Deletes role from the database when it's deleted from the guild."""
+        await self.bot.api_client.delete('bot/roles/' + str(role.id))
```
This is the prevalent style in this cog package. I can change all entries or keep the ones I've added consistent.

I figured it was better to keep the got blame mostly intact, since a single `+` operation is not something that's unclear or in any way less performant. 

I can change all the occurences, though. I'm pretty neutral on it.

SebastiaanZ replied:
I've changed all the occurrences to f-strings.

SebastiaanZ replied:
I've changed all the occurrences to f-strings.

SebastiaanZ commented on bot/cogs/sync/cog.py:
```diff
@@ -30,54 +30,68 @@ def __init__(self, bot):
     async def on_ready(self):
+        """Syncs the roles/users of the guild with the database."""
         guild = self.bot.get_guild(self.SYNC_SERVER_ID)
         if guild is not None:
```
I've not touched this part of the code, only added a docstring. I can rewrite it, but it's not really something I would consider to be falling within the scope of Kaizen, since it changes logic and will make the code unfamiliar to the original author.

SebastiaanZ commented:
> Incorrect types and missing mention of third value in docstring descriptions:
> 
> https://github.com/python-discord/bot/blob/bb6e9a050d14c375243fecb15f8fdffe7ad67787/bot/cogs/sync/syncers.py#L27

Changed.

> https://github.com/python-discord/bot/blob/bb6e9a050d14c375243fecb15f8fdffe7ad67787/bot/cogs/sync/syncers.py#L115

This was is correct; the user syncers only return two sets, not three.

> https://github.com/python-discord/bot/blob/bb6e9a050d14c375243fecb15f8fdffe7ad67787/bot/cogs/sync/syncers.py#L129

Already correct, see previous.

> https://github.com/python-discord/bot/blob/bb6e9a050d14c375243fecb15f8fdffe7ad67787/bot/cogs/sync/syncers.py#L129

This link points to the same line as the previous link. Did you mean another line number?

> These are missing a return type:
> 
> https://github.com/python-discord/bot/blob/bb6e9a050d14c375243fecb15f8fdffe7ad67787/bot/cogs/sync/syncers.py#L49
>
> https://github.com/python-discord/bot/blob/bb6e9a050d14c375243fecb15f8fdffe7ad67787/bot/cogs/sync/syncers.py#L167

Changed both.

lemonsaurus commented on bot/cogs/sync/cog.py:
```diff
@@ -26,58 +26,72 @@ class Sync:
+                total_created, total_updated, total_deleted = await syncer(self.bot, guild)
+                if total_deleted is None:
+                    log.info(
+                        "`%s` syncer finished, created `%d`, updated `%d`.",
```
I thought these were changed to f-strings in a previous commit but maybe that was something else?

I'm generally not a fan of this interpolation style, if we don't have a good reason for using it I'd prefer we switch it out with something more readable for everything in this file.

There _are_ legitimate reasons for using this (for example if you're interpolating something very heavy and you're doing it at loglevel info and we're running loglevel error in production, that heavy thing will never be evaluated), but I don't think they apply to any of these.

SebastiaanZ replied:
I must have fucked somewhere or did not actually change the code after the agreement we had on it.

SebastiaanZ replied:
As far as I can tell, I just completely forgot to change the code after our discussion in dev-core.
